File: ui/android/wuffittracker/src/com/androidworks/navsys/wuffit/content/WuffITSMSReceiver.java
package com.androidworks.navsys.wuffit.content;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class WuffITSMSReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Intent serviceIntent = new Intent("com.androidworks.navsys.wuffit.SERVICE_SMS_HANDLER");
        serviceIntent.putExtra("SMS_BUNDLE", intent.getExtras());
        context.startService(serviceIntent);
    }
}

File: ui/android/wuffittracker/src/main/kotlin/com/androidworks/navsys/wuffit/content/WuffITSMSReceiver.kt
package com.androidworks.navsys.wuffit.content

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.androidworks.navsys.wuffit.service.SMSHandler

class WuffITSMSReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val serviceIntent = Intent(context, SMSHandler::class.java).apply {
            action = "com.androidworks.navsys.wuffit.SERVICE_SMS_HANDLER"
            putExtra("SMS_BUNDLE", intent.extras)
        }
        context.startService(serviceIntent)
    }
}
